src/node/quoteverification.h

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#ifdef GET_QUOTE
+
+#  include "enclave/oe_shim.h"
+#  include "entities.h"
+#  include "networktables.h"
+
+#  include <vector>
+
+namespace ccf
+{
+  enum QuoteVerificationResult : uint32_t
+  {
+    VERIFIED = 0,
+    FAIL_VERIFY_OE,
+    FAIL_VERIFY_CODE_ID_RETIRED,
+    FAIL_VERIFY_CODE_ID_NOT_FOUND,
+    FAIL_VERIFY_INVALID_HASH,
+  };
+
+  class QuoteVerifier
+  {
+  private:
+    static QuoteVerificationResult verify_enclave_measurement(
+      Store::Tx& tx,
+      NetworkTables& network,
+      Cert& cert,
+      oe_report_t& parsed_quote)
+    {
+      // Verify enclave measurement
+      auto codeid_view = tx.get_view(network.code_id);
+      CodeStatus code_id_status = CodeStatus::UNKNOWN;
+
+      codeid_view->foreach([&parsed_quote, &code_id_status](
```

Comment:
We should add a `TODO` here to change this once the `CodeID` table becomes a digest -> status mapping.

Reply:
I already have a conflict on my other branch